common/src/View/MapDocument.cpp

```diff
@@ -1401,34 +1386,49 @@ namespace TrenchBroom {
             return submitAndStore(FindPlanePointsCommand::findPlanePoints());
         }
 
-        MapDocument::MoveVerticesResult MapDocument::moveVertices(const std::map<vm::vec3, std::set<Model::Brush*>>& vertices, const vm::vec3& delta) {
-            MoveBrushVerticesCommand::Ptr command = MoveBrushVerticesCommand::move(vertices, delta);
-            const bool success = submitAndStore(command);
-            const bool hasRemainingVertices = command->hasRemainingVertices();
+        MapDocument::MoveVerticesResult MapDocument::moveVertices(const std::map<vm::vec3, std::vector<Model::Brush*>>& vertices, const vm::vec3& delta) {
+            /*
+             * Don't auto this! We need the implicit conversion from std::unique_ptr<MoveBrushVerticesCommand> to
+             * std::unique_ptr<UndoableCommand> here, otherwise it happens when calling submitAndStore, which leads to
+             * the command object being deleted if the command fails. But we still need the command to check whether
+             * there were remaining vertices.
+             */
+            std::unique_ptr<UndoableCommand> command = MoveBrushVerticesCommand::move(vertices, delta);
```

Comment:
Yikes. Why does "otherwise it happens when calling submitAndStore, which leads to the command object being deleted if the command fails." happen if it were a std::unique_ptr<MoveBrushVerticesCommand>, but not when it's a std::unique_ptr<UndoableCommand>?

Reply:
fixed